src/mlpack/core/optimizers/pso/constriction_factor.hpp

```diff
@@ -23,6 +23,23 @@ namespace optimization {
 class ConstrictionFactor
 {
  public:
+  /**
+   * Initialize the constant values required for updating velocity.
+   * In this case, the constriction factor is computed.
+   *
+   * @param cognitiveAcceleration Cognitive acceleration of the particles.
+   * @param socialAcceleration Social acceleration of the particles.
+   */
+  void Initialize(const double& cognitiveAcceleration,
```

Comment:
@zoq I have added an `Initialize` method which computes the constriction factor. However, in case of inertia weight variant of PSO, this method takes dummy parameters. Maybe this could be avoided using SFINAE, i.e. by calling `velocityType.Initialize` only if the `ConstrictionFactor` template is provided. But this might require code duplication.

This code executes different functions based on the template parameter: https://stackoverflow.com/a/30440624

Reply:
I think the dummy parameter solution is just fine, SFINAE would be fine as well, but I think the current way is more readability and less complex, your call.

Reply:
Alright, I think the dummy parameter solution is good for now.

Regarding this PR, should I wait until the GSoC results are announced before this gets merged into master? Or, would it be fine to merge the basic PSO now? (assuming it doesn't require further changes)

Reply:
No need to wait, once this is ready we can merge it. I'll have to recheck the update step.